torchtext/prototype/models/t5/t5_transform.py

```diff
@@ -0,0 +1,85 @@
+from typing import Any, List
+
+import torch
+import torch.nn as nn
+import torchtext.transforms as T
+from torchtext.data.functional import load_sp_model
+from torchtext.functional import to_tensor
+from torchtext.utils import get_asset_local_path
+
+
+class T5Transform(nn.Module):
+    """
+    Transform for Sentence Piece tokenizer from pre-trained sentencepiece model in preparation for T5 model
+
+    Additiona details: https://github.com/google/sentencepiece
+
+    :param sp_model_path: Path to pre-trained sentencepiece model
+    :type sp_model_path: str
+    :param max_seq_len: Maximum sequence length accepted for inputs to T5 model
+    :type max_seq_len: int
+    :param eos_idx: End-of-sequence token id
+    :type eos_idx: int
+    :param padding_idx: Padding token id
+    :type padding_idx: int
+
+    Example
+        >>> from torchtext.prototype.models import T5Transform
+        >>> transform = T5Transform("spm_model", max_seq_len = 10, eos_idx = 1, padding_idx = 0)
+        >>> transform(["hello world", "attention is all you need!"])
+    """
+
+    def __init__(self, sp_model_path: str, max_seq_len: int, eos_idx: int, padding_idx: int):
+        super().__init__()
+        self.sp_model = load_sp_model(get_asset_local_path(sp_model_path))
+        self.max_seq_len = max_seq_len
+        self.eos_idx = eos_idx
+        self.padding_idx = padding_idx
+        self.pipeline = T.Sequential(T.Truncate(self.max_seq_len), T.AddToken(token=self.eos_idx, begin=False))
+
+    def forward(self, input: Any) -> Any:
```

Comment:
Is there a reason we specify the input as `Any` instead of `Union[str, List[str]]`?

Reply:
I initially had them typed, but noticed that [SentencePieceTokenizer](https://github.com/pytorch/text/blob/af03d7101c3c86aee9ecb0dd724bb0bf5b2b09da/torchtext/transforms.py#L56) had them as `Any` so deferred to that. I will revert it back.

Reply:
> Is there a reason we specify the input as `Any` instead of `Union[str, List[str]]`?

I guess @pmabbo13 might have followed what we did in our transform implementation where we always use `Any` as type for input. The reason for this is to ensure when transforms are combined in `SequentialTransform`, the overall transform is still scriptable. More details about this issue can be found  [here](https://github.com/pytorch/text/pull/1453#issue-1071329083). As for T5Transform, if we do not expect it to be used in `SequentialTransform` and treat it as a standalone one, I agree that we could just use the right annotation types as suggested above.